src/actions/MatrixActionCreators.js

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import dis from '../dispatcher';
+
+// TODO: migrate from sync_state to MatrixActions.sync so that more js-sdk events
+//       become dispatches in the same place.
+/**
+ * An action creator that will map a `sync` event to a MatrixActions.sync action,
```

Comment:
I've got to say, I think calling these things action creators as well as the things that `asyncAction` returns is confusing. Why not just "Create a `MatrixActions.sync` action which represents a MatrixClient `sync` event", or sth like that?

Reply:
WFM